File: frontend/src/types/order.ts
export type OrderSide = 'BUY' | 'SELL'

export type OrderStatus = 'PENDING' | 'ACCEPTED' | 'FILLED' | 'REJECTED'

export interface Account {
  id: number
  cashBalance: number
  lockedMargin: number
  availableBalance: number
}

export interface Instrument {
  symbol: string
  name: string
  lastPrice: number
}

export interface Quote {
  symbol: string
  name: string
  lastPrice: number
  volume: number
  updatedAt: string
}

export interface Order {
  id: string
  accountId: number
  symbol: string
  instrumentName: string
  side: OrderSide
  quantity: number
  price: number
  status: OrderStatus
  createdAt: string
}

export interface CreateOrderRequest {
  symbol: string
  side: OrderSide
  quantity: number
  price: number
}

File: backend/src/main/java/com/bellwin/stock/domain/Order.java
package com.bellwin.stock.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import java.time.Instant;
import java.util.UUID;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "orders")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Order {

    @Id
    @Column(length = 36)
    private String id;

    @Column(nullable = false)
    private Long accountId;

    @Column(nullable = false, length = 12)
    private String symbol;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 8)
    private OrderSide side;

    @Column(nullable = false)
    private int quantity;

    @Column(nullable = false)
    private long price;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 16)
    private OrderStatus status;

    @Column(nullable = false)
    private Instant createdAt;

    @Version
    private Long version;

    @Builder
    public Order(Long accountId, String symbol, OrderSide side, int quantity, long price) {
        this.id = UUID.randomUUID().toString();
        this.accountId = accountId;
        this.symbol = symbol;
        this.side = side;
        this.quantity = quantity;
        this.price = price;
        this.status = OrderStatus.PENDING;
        this.createdAt = Instant.now();
    }

    public long orderAmount() {
        return (long) quantity * price;
    }

    public void markAccepted() {
        this.status = OrderStatus.ACCEPTED;
    }

    public void markFilled() {
        this.status = OrderStatus.FILLED;
    }

    public void markRejected() {
        this.status = OrderStatus.REJECTED;
    }
}
